src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/resources/RuntimeHelper.java.template

```diff
@@ -69,11 +70,21 @@ public class RuntimeHelper {
         return seg.withAccessModes(seg.accessModes() &  ~MemorySegment.CLOSE & ~MemorySegment.HANDOFF);
     }
 
+    private static final boolean isVaList(FunctionDescriptor desc) {
+        List<MemoryLayout> argLayouts = desc.argumentLayouts();
+        int size = argLayouts.size();
+        if (size > 0) {
+            MemoryLayout ml = argLayouts.get(size - 1);
+            return ml.equals(C_VA_LIST) && ml.name().orElse("").equals("__va_list_tag");
+        }
+        return false;
+    }
+
     public static final MethodHandle downcallHandle(LibraryLookup[] LIBRARIES, String name, String desc, FunctionDescriptor fdesc, boolean variadic) {
         return lookup(LIBRARIES, name).map(
                 addr -> {
                     MethodType mt = MethodType.fromMethodDescriptorString(desc, LOADER);
-                    return variadic ?
+                    return variadic || isVaList(fdesc)?
                         VarargsInvoker.make(addr, mt, fdesc) :
```

Comment:
I don't get this change - we don't want a vararg invoker for a valist call, right? The invoker will most surely not do the right thing here.

Also, the test seems to show that no Java varargs invocation is being applied (which is correct), so what is the reasons for the changes in RuntimeHelper?

Reply:
missng space between `variadic` and `?`